### ISSUE TITLE
GetAccountLedger()  Error deserializing data

### ISSUE THREAD

**Issue by RiveGauche** [JKorf/Kucoin.Net#18]
Could not cast or convert from System.String to Kucoin.Net.Model.KucoinAccountActivityContext.

**Comment by JKorf** [comment]
Can you have a look in the log output to see what the data returned is? According to the API docs it should be an object, not a string.

**Comment by Zebio21** [comment]
This function is deprecated but can be called :)
I've the same problem !

DATA : 
}"},{"id":"601f20cd27bf6900068c072e","currency":"BTC","amount":"0.00028771","fee":"0.00000028","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1612652749222,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"601f20cd564b7d0006ab7fed\",\"tradeId\":\"601f20cd182ace1b21848b59\"}"},{"id":"6017d221bacb8e00067c3841","currency":"BTC","amount":"0.00072965","fee":"0.00000072","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1612173857782,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"6017d221e0162d0006e124c6\",\"tradeId\":\"6017d2212e6dcc43c7ed41f6\"}"},{"id":"6017d1fabacb8e00067bf634","currency":"BTC","amount":"0.00000004","fee":"0","balance":"0","accountType":"TRADE","bizType":"Refunded Fees","direction":"in","createdAt":1612173818335,"context":"{\"orderId\":\"6017d1f9e5eae10006fa29e1\",\"tradeId\":\"6017d1f92e6dcc57d366a849\"}"},{"id":"6017d1fa86a31e0006a8a834","currency":"BTC","amount":"0.00000001","fee":"0","balance":"0","accountType":"TRADE","bizType":"Refunded Fees","direction":"in","createdAt":1612173818297,"context":"{\"orderId\":\"6017d1f9e5eae10006fa29e1\",\"tradeId\":\"6017d1f92e6dcc57d366a84a\"}"},{"id":"6017d1f96a870d00066d6d75","currency":"BTC","amount":"0.00066345","fee":"0.00000066","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1612173817924,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"6017d1f9e5eae10006fa29e1\",\"tradeId\":\"6017d1f92e6dcc57d366a84b\"}"},{"id":"6017d1f96a870d00066d6d76","currency":"BTC","amount":"0.00001928","fee":"0.00000001","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1612173817922,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"6017d1f9e5eae10006fa29e1\",\"tradeId\":\"6017d1f92e6dcc57d366a84a\"}"},{"id":"6017d1f96a870d00066d6d74","currency":"BTC","amount":"0.00004685","fee":"0.00000004","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1612173817922,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"6017d1f9e5eae10006fa29e1\",\"tradeId\":\"6017d1f92e6dcc57d366a849\"}"},{"id":"6015ececa2ef3d0006f747fd","currency":"BTC","amount":"0.00000019","fee":"0","balance":"0","accountType":"TRADE","bizType":"Convert to KCS","direction":"out","createdAt":1612049644087,"context":"{\"smallCurrencyExchangeId\":\"6015eceb14d5520006405d23\"}"},{"id":"600d75ec86a31e0006162c15","currency":"BTC","amount":"0.00099016","fee":"0.00000098","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1611494892482,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"600d75ec5c42ad00063c58e9\",\"tradeId\":\"600d75ec2e6dcc43c7e24e71\"}"},{"id":"600d743e9470d60006a50bca","currency":"BTC","amount":"0.00000002","fee":"0","balance":"0","accountType":"TRADE","bizType":"Refunded Fees","direction":"in","createdAt":1611494462313,"context":"{\"orderId\":\"600d743ef6b48b0006263040\",\"tradeId\":\"600d743e2e6dcc57d36092e9\"}"},{"id":"600d743ebacb8e0006e807d0","currency":"BTC","amount":"0.00002186","fee":"0.00000002","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1611494462148,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"600d743ef6b48b0006263040\",\"tradeId\":\"600d743e2e6dcc57d36092e9\"}"},{"id":"600d743ebacb8e0006e807d3","currency":"BTC","amount":"0.00048474","fee":"0.00000048","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1611494462147,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"600d743ef6b48b0006263040\",\"tradeId\":\"600d743e2e6dcc57d36092eb\"}"},{"id":"600d743ebacb8e0006e807d4","currency":"BTC","amount":"0.00048367","fee":"0.00000048","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1611494462147,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"600d743ef6b48b0006263040\",\"tradeId\":\"600d743e2e6dcc57d36092ea\"}"},{"id":"600792232c7ae70006d55f6d","currency":"BTC","amount":"0.00024467","fee":"0.00000024","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1611108899853,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"6007922374d77c00061e8c83\",\"tradeId\":\"600792232e6dcc43c7db2621\"}"},{"id":"6007920e8af7d7000678f5d1","currency":"BTC","amount":"0.00024473","fee":"0.00000024","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1611108879005,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"6007920e525b6a0006251f61\",\"tradeId\":\"6007920e182ace1b216c6311\"}"},{"id":"60016b4b35c80400069091be","currency":"BTC","amount":"0.00052174","fee":"0.00000052","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1610705739056,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"60016b4af3a49f0006bb34d3\",\"tradeId\":\"60016b4a2e6dcc43c7d38f0e\"}"},{"id":"60016b29b812c20006c5002e","currency":"BTC","amount":"0.0005212","fee":"0.00000052","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1610705705704,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"60016b29e0ad810006aec047\",\"tradeId\":\"60016b292e6dcc57d3599706\"}"},{"id":"5ffc3e0125587e0006b17b7a","currency":"BTC","amount":"0.00000026","fee":"0","balance":"0","accountType":"TRADE","bizType":"Refunded Fees","direction":"in","createdAt":1610366465849,"context":"{\"orderId\":\"5ffc3e007dfc5300061fabcd\",\"tradeId\":\"5ffc3e012e6dcc43c7cba48d\"}"},{"id":"5ffc3e0128a53900078c3f80","currency":"BTC","amount":"0.00026239","fee":"0.00000026","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1610366465486,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5ffc3e007dfc5300061fabcd\",\"tradeId\":\"5ffc3e012e6dcc43c7cba48d\"}"},{"id":"5ffc3da10f811c00062c8c78","currency":"BTC","amount":"0.00000026","fee":"0","balance":"0","accountType":"TRADE","bizType":"Refunded Fees","direction":"in","createdAt":1610366369353,"context":"{\"orderId\":\"5ffc3da078940e0006a0a733\",\"tradeId\":\"5ffc3da0182ace1b2160aa80\"}"},{"id":"5ffc3da1cd88d90006b45153","currency":"BTC","amount":"0.00026241","fee":"0.00000026","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1610366369132,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"5ffc3da078940e0006a0a733\",\"tradeId\":\"5ffc3da0182ace1b2160aa80\"}"},{"id":"5ffb141792dd010006380f83","currency":"BTC","amount":"0.00000051","fee":"0","balance":"0","accountType":"TRADE","bizType":"Convert to KCS","direction":"out","createdAt":1610290199828,"context":"{\"smallCurrencyExchangeId\":\"5ffb14176a2c7f00064f6b7c\"}"},{"id":"5ff7ff7c9b2d2800063ef6d0","currency":"BTC","amount":"0.00050284","fee":"0.0000005","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1610088316291,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5ff7ff7cb7e9e80006f9febc\",\"tradeId\":\"5ff7ff7c2e6dcc43c7c61cf4\"}"},{"id":"5ff7ff657100d7000691a278","currency":"BTC","amount":"0.00050037","fee":"0.0000005","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1610088293377,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5ff7ff65eaac410006a69727\",\"tradeId\":\"5ff7ff652e6dcc57d3594c35\"}"},{"id":"5ff3b3b87100d700063dd739","currency":"BTC","amount":"0.00085119","fee":"0.00000085","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1609806776920,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5ff3b3b876d406000632e167\",\"tradeId\":\"5ff3b3b82e6dcc43c7c0b44f\"}"},{"id":"5ff1ea778b10400006c2d37a","currency":"BTC","amount":"0.00018046","fee":"0.00000018","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1609689719580,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"5ff11566ded1c00006b872a5\",\"tradeId\":\"5ff1ea77182ace1b2153dbef\"}"},{"id":"5ff153b21548210006849712","currency":"BTC","amount":"0.00000654","fee":"0","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1609651122741,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"5ff11566ded1c00006b872a5\",\"tradeId\":\"5ff153b2182ace1b2152fa68\"}"},{"id":"5ff114de429b2800065e8fe9","currency":"BTC","amount":"0.00066714","fee":"0.00000066","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1609635038410,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5ff114de20346f0006eeed24\",\"tradeId\":\"5ff114de2e6dcc57d3591c0b\"}"},{"id":"5fe3cf3571e4410006459d0b","currency":"BTC","amount":"0.02128165","fee":"0.00002126","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1608765237635,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5fe3cf35fb0b1700063f2ddc\",\"tradeId\":\"5fe3cf352e6dcc43c7b12d53\"}"},{"id":"5fe3cefe972924000678f658","currency":"BTC","amount":"0.02128162","fee":"0","balance":"0","accountType":"TRADE","bizType":"Transfer","direction":"in","createdAt":1608765182573,"context":null},{"id":"5fe27771b4d2ba0007511869","currency":"BTC","amount":"0.00098891","fee":"0.00000098","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1608677233359,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5fe277711edf700006732557\",\"tradeId\":\"5fe277712e6dcc43c7aff788\"}"},{"id":"5fe27771b4d2ba0007511866","currency":"BTC","amount":"0.00001201","fee":"0.00000001","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1608677233359,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5fe277711edf700006732557\",\"tradeId\":\"5fe277712e6dcc43c7aff787\"}"},{"id":"5fe2771d41aff800063888b0","currency":"BTC","amount":"0.00000825","fee":"0","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1608677149614,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fe2771d20346f00063a5309\",\"tradeId\":\"5fe2771d2e6dcc57d35873bf\"}"},{"id":"5fe2771d41aff800063888af","currency":"BTC","amount":"0.00096715","fee":"0.00000096","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1608677149614,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fe2771d20346f00063a5309\",\"tradeId\":\"5fe2771d2e6dcc57d35873c0\"}"},{"id":"5fe2771d41aff800063888ae","currency":"BTC","amount":"0.00002555","fee":"0.00000002","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1608677149614,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fe2771d20346f00063a5309\",\"tradeId\":\"5fe2771d2e6dcc57d35873be\"}"},{"id":"5fd755e8dea1330006679af9","currency":"BTC","amount":"0.00193816","fee":"0.00000193","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1607947752161,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5fd755e8fbe287000753050a\",\"tradeId\":\"5fd755e82e6dcc43c7a6e3f6\"}"},{"id":"5fd755b89434e00006dba29e","currency":"BTC","amount":"0.00043969","fee":"0.00000044","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607947704989,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"5fd755b8a96d2100063af81b\",\"tradeId\":\"5fd755b8182ace1b2138c4d1\"}"},{"id":"5fd755a19434e00006db83ab","currency":"BTC","amount":"0.00052176","fee":"0.00000052","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607947681609,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd755a18e2b00000636a49b\",\"tradeId\":\"5fd755a12e6dcc57d356a8ae\"}"},{"id":"5fd755a19434e00006db83ac","currency":"BTC","amount":"0.00097666","fee":"0.00000097","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607947681609,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd755a18e2b00000636a49b\",\"tradeId\":\"5fd755a12e6dcc57d356a8af\"}"},{"id":"5fd6e1818677aa000687d9ab","currency":"BTC","amount":"0.00300575","fee":"0.000003","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1607917953863,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5fd6e1811f8ea80006514d72\",\"tradeId\":\"5fd6e1812e6dcc43c7a68ac3\"}"},{"id":"5fd6e1818677aa000687d9aa","currency":"BTC","amount":"0.00003613","fee":"0.00000003","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"out","createdAt":1607917953863,"context":"{\"symbol\":\"NRG-BTC\",\"orderId\":\"5fd6e1811f8ea80006514d72\",\"tradeId\":\"5fd6e1812e6dcc43c7a68ac2\"}"},{"id":"5fd6e14c9f59520006e77227","currency":"BTC","amount":"0.00065512","fee":"0.00000065","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607917900745,"context":"{\"symbol\":\"PIVX-BTC\",\"orderId\":\"5fd6e14ca96d2100061b0429\",\"tradeId\":\"5fd6e14c182ace1b21386414\"}"},{"id":"5fd6e137b9cb910006a22880","currency":"BTC","amount":"0.00074793","fee":"0.00000074","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607917879944,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd6e137fbe28700072f17f3\",\"tradeId\":\"5fd6e1372e6dcc57d3569040\"}"},{"id":"5fd6e137b9cb910006a2287d","currency":"BTC","amount":"0.00005853","fee":"0.00000005","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607917879944,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd6e137fbe28700072f17f3\",\"tradeId\":\"5fd6e1372e6dcc57d356903d\"}"},{"id":"5fd6e137b9cb910006a2287f","currency":"BTC","amount":"0.00099192","fee":"0.00000099","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607917879939,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd6e137fbe28700072f17f3\",\"tradeId\":\"5fd6e1372e6dcc57d356903f\"}"},{"id":"5fd6e137b9cb910006a2287c","currency":"BTC","amount":"0.00009155","fee":"0.00000009","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607917879938,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd6e137fbe28700072f17f3\",\"tradeId\":\"5fd6e1372e6dcc57d356903c\"}"},{"id":"5fd6e137b9cb910006a2287e","currency":"BTC","amount":"0.00024592","fee":"0.00000024","balance":"0","accountType":"TRADE","bizType":"Exchange","direction":"in","createdAt":1607917879938,"context":"{\"symbol\":\"ZEL-BTC\",\"orderId\":\"5fd6e137fbe28700072f17f3\",\"tradeId\":\"5fd6e1372e6dcc57d356903e\"}"}]}}

MESSAGE :
Deserialize JsonSerializationException: Error converting value "{"symbol":"NRG-BTC","orderId":"60212d97e9efb1000671f28c","tradeId":"60212d972e6dcc5467e6f93e"}" to type 'Kucoin.Net.Objects.KucoinAccountActivityContext'. Path 'data.items[0].context', line 1, position 403.

**Comment by Neriondiel** [comment]
Hi, 
I have the same problem here. And like Zebio21 said, this endpoint seem to be deprecated 😢  
https://docs.kucoin.com/#get-account-ledgers-deprecated
There is another endpoint and we are no longer forced to give one parameter
https://docs.kucoin.com/#get-account-ledgers
Maybe an update to the new endpoint may help ? 😄

**Comment by Zebio21** [comment]
Hello, 

I've downloaded the full source and start to work on KuCoin implementation. 

I've tagued the old fonction to obsolete and create the new one.

For this problem, the API returns an object in a string...

With a converter who parse this string, it's OK.

The other problem is that the context object isn't described and few possibles properties are missing.

I'm not fluent with github, how can i purpose my modifications ?
Or can i join the project ?

I'll work soon on the Crex24 implementation too 👍
